tidy3d/web/webapi.py

```diff
@@ -579,27 +529,35 @@ def load(
 
     Returns
     -------
-    :class:`.SimulationData`
+    :class:`.StubData`
         Object containing simulation data.
     """
 
+    task = get_info(task_id)
+
     if not os.path.exists(path) or replace_existing:
         download(task_id=task_id, path=path, verbose=verbose, progress_callback=progress_callback)
 
     if verbose:
         console = get_logging_console()
         console.log(f"loading SimulationData from {path}")
 
-    sim_data = SimulationData.from_file(path)
+    if TaskType.FDTD == task.taskType or TaskType.NORMAL == task.taskType:
```

Comment:
it might be nice to implement these as an `abstract method` in the `StubData` for example a `def postprocess` and then implement in the subclasses?

Reply:
use function 'load_data(task_type: TaskType, path: str)'